src/main/webapp/js/selectScript.js

```diff
@@ -14,40 +14,40 @@
 export const CollegePlanner = (() => {
   let selected = [];  // Courses selected by the user
   let courses = [];   // List with all courses
+  let courseIdNameMap = {};  // JSON mapping of course ID {string} to course name {string}
 
   /**
    * Gets courses from /courselist servlet to populate dropdown list
    */
   async function getOptions() {
     const response = await fetch('/api/courses');
-    courseList = await response.json();
+    const courseList = await response.json();
     const courseContainer = document.getElementById('courses');
     courseContainer.innerHTML = '';
-    addOption('Select a Course', courseContainer, /*shouldSetValue=*/ false);
+    // Add default option to course list
+    const option = document.createElement('option');
+    option.innerText = 'Select a Course';
+    option.selected = true;
+    option.hidden = true;
+    courseContainer.appendChild(option);
+    // Add each course to course list
     const coursesDetailed = courseList.courses_detailed;
-    coursesDetailed.forEach(
-        course =>
-            addOption(course.name, courseContainer, /*shouldSetValue=*/ true));
+    coursesDetailed.forEach(course => addOption(course, courseContainer));
   }
 
   /**
    * Creates options in select list
-   * @param {string} course The name of the course to add to the dropdown
+   * @param {Object} course The JSON Object for the course to add to the
+   *     dropdown
    * @param {string} container The id name of the container you want to add
    *     options to
-   * @param {boolean} shouldSetValue Whether to set the value of the option to
-   *     the string
    */
-  function addOption(course, courseContainer, shouldSetValue) {
+  function addOption(course, courseContainer) {
+    courseIdNameMap[`${course.course_id}`] = course.name;
```

Comment:
no need for string interpolation: `courseIdNameMap[course.course_id]` is sufficient